kernel/os/src/arch/cortex_m33/m33/HAL_CM33.s

```diff
@@ -183,6 +183,11 @@ PendSV_Handler:
         STR     R12,[R1,#0]             /* Update stack pointer in current task */
         STR     R2,[R3]                 /* g_current_task = highest ready */
 
+        LDR     R12,[R2,#4]             /* get stack top of task we will start */
+        LDRH    R3,[R2,#8]              /* calculate stack size */
+        LSL     R3,R3,#2
```

Comment:
LSL R3, R3, #2
SUB R12, R12, R3
could be change to
SUB R12, R12, R3, LSL #2
saving 4 bytes

If on the other hand os_task_t stored stack bottom instead of stack top (what could be understood from @wes3 comment), code would be even shorter, and coverity tool would not issue false positive issues about out of bound access somewhere during task creation.

Reply:
I would store it :-)